### DISCUSSION TITLE
Assignment 4 is assigned!!!

### DISCUSSION THREAD

**Discussion by enigma037** [Orchid-DOTNET/BscCSIT-2023#44] [Announcements]
@Orchid-DOTNET/sectiona-2023 @Orchid-DOTNET/sectionb-2023 

https://github.com/orgs/Orchid-DOTNET/projects/7/views/1 

<img width="419" alt="image" src="https://user-images.githubusercontent.com/10417012/227873832-89223324-aa38-4c6d-8bcf-04c9161b1711.png">

**Comment by ranisha007** [comment]
On Mon, 27 Mar 2023 at 13:24, enigma037 ***@***.***> wrote:

> @Orchid-DOTNET/sectiona-2023
> <https://github.com/orgs/Orchid-DOTNET/teams/sectiona-2023>
> @Orchid-DOTNET/sectionb-2023
> <https://github.com/orgs/Orchid-DOTNET/teams/sectionb-2023>
>
> https://github.com/orgs/Orchid-DOTNET/projects/7/views/1
>
> [image: image]
> <https://user-images.githubusercontent.com/10417012/227873097-fcbe245c-cc54-495c-bc2d-ec6ea9e6dccb.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Orchid-DOTNET/discussions/44>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6XUSNJWWZBU76S4DWXV7W3W6E72HANCNFSM6AAAAAAWIZTS2E>
> .
> You are receiving this because you are on a team that was mentioned.Message
> ID: ***@***.***>
>

﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Assignment4;

class Calculator: cal1, cal2, cal3, cal4
{
    public int add(int x, int y)
    {
        return x + y;
    }
    public int sub(int x, int y)
    {
        return x - y;
    }
    public int mul(int x, int y)
    {
        return x * y;
    }
    public double div(int x, int y)
    {
        return x/y;
    }
}

﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignment4
{
    interface cal1
    {
        public int add(int x, int y);
    }
    interface cal2
    {
        public int sub(int x, int y);
    }
    interface cal3
    {
        public int mul(int x, int y);
    }
    interface cal4
    {
        public  double div(int x, int y);
    }
}

﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignment3
{
    public class Employee
    {
        public int id;
        public string name;
        public int age;

        public Employee (int id, string name, int age)
        {
            this.id = id;
            this.name = name;
            this.age = age;
        }

    }

    class EmployeeDetails
    {
        public void empDetails()
        {
            List<Employee> EmployeeList = new List<Employee>()
            {
            new Employee(01, "John Wick", 55),
            new Employee(02, "Matt Murdock", 32),
            new Employee(03, "Peter Parker", 20)
            };

            var result = from emp in EmployeeList
                         where emp.age > 30
                         orderby emp.id
                         select emp;
            foreach (var emp in result)
            {
                Console.WriteLine("ID: {0}", emp.id);
                Console.WriteLine("Name: {0}", emp.name);
                Console.WriteLine("Age: {0}", emp.age);
            }
        }

    }
}

using System;

namespace Assignment1{
    class EditString{
        public static void String_Edit(){
            string var ="dotnet Centric Computing";
            string[] myVar=var.Split();
            foreach(string b in myVar)
            {
                Console.WriteLine(b);
            }

        }
    }
}
﻿using System;
using System.IO;
using System.Text;

class filexercise3
{
    public static void Main()
    {
        string fileName = @"mytest.txt";
        int count;

        try
        {
            // Delete the file if it exists.
            if (File.Exists(fileName))
            {
                File.Delete(fileName);
            }
            Console.Write("\n\n Count the number of lines in a file :\n");
            Console.Write("------------------------------------------\n");
            // Create the file.
            using (StreamWriter fileStr = File.CreateText(fileName))
            {
                fileStr.WriteLine(" welcome ");
                fileStr.WriteLine(" to ");
                fileStr.WriteLine(" orchid ");
                fileStr.WriteLine(" international ");
                fileStr.WriteLine(" college ");
                fileStr.WriteLine(" kathmandu ");
                fileStr.WriteLine(" gaushala ");
                fileStr.WriteLine(" bijay chowk ");
            }
            using (StreamReader sr = File.OpenText(fileName))
            {
                string s = "";
                count = 0;
                Console.WriteLine(" Here is the content of the file mytest.txt : ");
                while ((s = sr.ReadLine()) != null)
                {
                    Console.WriteLine(s);
                    count++;
                }
                Console.WriteLine("");
            }
            Console.Write(" The number of lines in  the file {0} is : {1} \n\n", fileName, count);
        }
        catch (Exception MyExcep)
        {
            Console.WriteLine(MyExcep.ToString());
        }
    }
}

﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignment5
{
    internal class exceptionHandling
    {
        public void div(int x, int y)
        {
             try
            {
                double result = x / y;
                Console.WriteLine("{0}/{1} = {2}", x, y, result);
            }

            catch(DivideByZeroException) {
                Console.WriteLine("Exception Caught: {0} was divided by 0", x);
            }


        }
    }
}
